elements/pl-dropdown/pl-dropdown.py

```diff
@@ -0,0 +1,195 @@
+import prairielearn as pl
+import lxml.html
+import random
+import chevron
+from enum import Enum
+
+WEIGHT_DEFAULT = 1
+BLANK_ANSWER = ' '
+BLANK_DEFAULT = True
+SORT_DEFAULT = 'fixed'
+NO_ANSWER_SELECTED = '<No answer selected>'
+
+
+class SortTypes(Enum):
+    RANDOM = 'random'
+    ASCEND = 'ascend'
+    DESCEND = 'descend'
+    FIXED = 'fixed'
+
+
+def get_options(element, data):
+    options = []
+    for child in element:
+        if child.tag in ['pl-answer']:
+            pl.check_attribs(child, required_attribs=[], optional_attribs=['correct'])
+            child_html = pl.inner_html(child).strip()
+            options.append(child_html)
+    return options
+
+
+def get_solution(element, data):
+    solution = []
+    for child in element:
+        if child.tag in ['pl-answer']:
+            pl.check_attribs(child, required_attribs=[], optional_attribs=['correct'])
+            is_correct = pl.get_boolean_attrib(child, 'correct', False)
+            child_html = pl.inner_html(child).strip()
+            if is_correct:
+                solution.append(child_html)
+
+    if len(solution) > 1:
+        raise Exception('Multiple correct answers were set')
+
+    return solution[0]
+
+
+def prepare(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    pl.check_attribs(element, required_attribs=['answers-name'], optional_attribs=['blank', 'weight', 'sort'])
+    answers_name = pl.get_string_attrib(element, 'answers-name')
+
+    # Get answer from pl-answer if implemented
+    data['correct_answers'][answers_name] = get_solution(element, data)
+
+    # Get answer from server.py if pl-answer not implemented
+    if answers_name is None:
+        data['correct_answers'][answers_name] = answers_name
```

Comment:
The code above looks suspicious to me! 😄

Reply:
Yes, thank you. Removed.